test/globalErr.js

```diff
@@ -108,4 +108,112 @@ describe('globalErr', function () {
+      // validates that it has the expected error message.
+      it('should throw if "Getter" is in the stack trace', function () {
+        err(function () {
+          err(function myGetter () {
+            throw Error('my stack trace contains a fake implementation frame');
+          });
+        }, /implementation frames not properly filtered from stack trace/, true);
+      });
+
+      // The inner `err` executes `myWrapper` and catches the thrown error. But
+      // then it sees the word "Wrapper" in the stack trace of the thrown error,
+      // so it thinks that implementation frames weren't properly filtered out,
+      // so it throws a new error. The outer `err` catches that new error and
+      // validates that it has the expected error message.
+      it('should throw if "Wrapper" is in the stack trace', function () {
+        err(function () {
+          err(function myWrapper () {
+            throw Error('my stack trace contains a fake implementation frame');
+          });
+        }, /implementation frames not properly filtered from stack trace/, true);
+      });
+
+      // The inner `err` executes `assert` and catches the thrown error. But
+      // then it sees the word "assert" in the stack trace of the thrown error,
+      // so it thinks that implementation frames weren't properly filtered out,
+      // so it throws a new error. The outer `err` catches that new error and
+      // validates that it has the expected error message.
+      it('should throw if "assert" is in the stack trace', function () {
+        err(function () {
+          err(function assert () {
+            throw Error('my stack trace contains a fake implementation frame');
+          });
+        }, /implementation frames not properly filtered from stack trace/, true);
+      });
+
+      // `err` executes `safeFnName` and catches the thrown error. Since there
+      // aren't any implementation frames in the stack trace of the thrown error,
+      // it shouldn't throw a new error (which would cause this test to fail).
+      it('shouldn\'t throw if "Getter", "Wrapper", "assert" aren\'t in the stack trace', function () {
+        err(function safeFnName () {
+          throw Error('my stack trace doesn\'t contain implementation frames');
+        });
+      });
+    });
+
+    describe('truthy', function () {
+      // `err` executes `myGetter` and catches the thrown error. Since
+      // `skipStackTest` is truthy, it shouldn't see the word "Getter" in the
+      // stack trace of the thrown error, and thus shouldn't throw a new error
+      // (which would cause this test to fail).
+      it('shouldn\'t throw if "Getter" is in the stack trace', function () {
+        err(function myGetter () {
+          throw Error('my stack trace contains a fake implementation frame');
+        }, undefined, true);
+      });
+
+      // `err` executes `myWrapper` and catches the thrown error. Since
```

Comment:
These comments are almost all the same, I think it adds a lot of noise to the code, especially because one of them would already be enough, but even more because now that we've got more descriptive error messages and awesome test case descriptions that already do the job of conveying the message we want it to.